projects/opendr_ws/src/perception/scripts/facial_emotion_estimation.py

```diff
@@ -0,0 +1,203 @@
+from opendr.perception.facial_expression_recognition import FacialEmotionLearner
+from opendr.perception.facial_expression_recognition import image_processing
+from opendr.perception.facial_expression_recognition import ESR
+
+
+class FacialEmotionEstimationNode:
+    def __init__(self,
+                 input_rgb_image_topic="/usb_cam/image_raw",
+                 output_emotions_topic="/opendr/facial_emotion",
+                 output_emotions_description_topic="/opendr/facial_emotion_estimation_description",
+                 device="cuda"):
+        """
+        Creates a ROS Node for facial emotion estimation
+        :param input_rgb_image_topic: Topic from which we are reading the input image
+        :type input_rgb_image_topic: str
+        :param output_emotions_topic: Topic to which we are publishing the facial emotion results
+        (if None, we are not publishing the info)
+        :type output_emotions_topic: str
+        :param output_emotions_description_topic: Topic to which we are publishing the description of the estimated
+        facial emotion (if None, we are not publishing the description)
+        :type output_emotions_description_topic: str
+        :param device: device on which we are running inference ('cpu' or 'cuda')
+        :type device: str
+        """
+
+        # Set up ROS topics and bridge
+        self.input_rgb_image_topic = input_rgb_image_topic
+        self.bridge = ROSBridge()
+
+        if output_emotions_topic is not None:
+            self.hypothesis_publisher = rospy.Publisher(output_emotions_topic, ObjectHypothesis, queue_size=1)
+        else:
+            self.hypothesis_publisher = None
+
+        if output_emotions_description_topic is not None:
+            self.string_publisher = rospy.Publisher(output_emotions_description_topic, String, queue_size=1)
+        else:
+            self.string_publisher = None
+
+        # Initialize the facial emotion estimator
+        self.facial_emotion_estimator = FacialEmotionLearner(device=device, batch_size=2,
+                                                             ensemble_size=9,
+                                                             name_experiment='esr_9')
+
+        URL_PATH = OPENDR_SERVER_URL + "perception/ensemble_based_cnn"
+        model_saved_path = self.facial_emotion_estimator.download(self, path=None, mode="pretrained", url=URL_PATH)
+
+        self.facial_emotion_estimator.load(self, ensemble_size=9, path_to_saved_network=model_saved_path,
+                                           file_name_base_network="Net-Base-Shared_Representations.pt",
+                                           file_name_conv_branch="Net-Branch_{}.pt", fix_backbone=True)
+
+    def listen(self):
+        """
+        Start the node and begin processing input data
+        """
+        rospy.init_node('opendr_facial_emotion_estimation', anonymous=True)
+        rospy.Subscriber(self.input_image_topic, ROS_Image, self.callback)
```

Comment:
```suggestion
        rospy.Subscriber(self.input_rgb_image_topic, ROS_Image, self.callback)
```

Reply:
also, do we need to set queue and buffer size to avoid delays?

Reply:
Also fixed this @negarhdr .